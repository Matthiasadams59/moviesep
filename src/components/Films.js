import React, {Component} from 'react'
import List from '@material-ui/core/List';
import Typography from '@material-ui/core/Typography';
import axios from 'axios'
import Film from './Film'

class Films extends Component {
    constructor(props) {
        super(props)

        this.state = {
            films: [],
        }
    }

    getFilms = (e) => {
        axios
        .get("http://localhost:8080/film/")
        .then( response => {
            this.setState({
                films: response.data,
            })
        })
        .catch(error => this.setState({ error }))
    }

    componentDidMount = () => {
        this.getFilms();
    }

    render() {

        return(
            <div>
                <Typography variant="h2">FILMS LIST</Typography>
                    <div>   
                        <List style={{display: 'flex', flexWrap: 'wrap', justifyContent: 'space-around',overflow: 'hidden'}}>
                            {
                                this.state.films.map((film, index) => (
                                    <Film key={index} film={film}/>                                    
                                )
                            )}   
                        </List>
                    </div>
            </div>
        )
    }
}

export default Films