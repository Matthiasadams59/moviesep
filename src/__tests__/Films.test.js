import React from 'react';
import { shallow } from "enzyme";
import Films from './../components/Films';

describe("<Films /> component", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Films />);
    })

    it('renders correctly', () => {
        expect(wrapper).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });

    it('initializes the `state` with an empty list of films', () => {
        expect(wrapper.state().films).toEqual([]);
    });

    it('ComponentDidMount', () => {
        wrapper.instance().componentDidMount();
        expect(wrapper.state().films.length).toBe(0);
      })
})


