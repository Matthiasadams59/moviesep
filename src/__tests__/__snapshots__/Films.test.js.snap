// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Films /> component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Films />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Typography)
          variant="h2"
        >
          FILMS LIST
        </WithStyles(Typography)>,
        <div>
          <WithStyles(List)
            style={
              Object {
                "display": "flex",
                "flexWrap": "wrap",
                "justifyContent": "space-around",
                "overflow": "hidden",
              }
            }
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "FILMS LIST",
          "variant": "h2",
        },
        "ref": null,
        "rendered": "FILMS LIST",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(List)
            style={
              Object {
                "display": "flex",
                "flexWrap": "wrap",
                "justifyContent": "space-around",
                "overflow": "hidden",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [],
            "style": Object {
              "display": "flex",
              "flexWrap": "wrap",
              "justifyContent": "space-around",
              "overflow": "hidden",
            },
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            variant="h2"
          >
            FILMS LIST
          </WithStyles(Typography)>,
          <div>
            <WithStyles(List)
              style={
                Object {
                  "display": "flex",
                  "flexWrap": "wrap",
                  "justifyContent": "space-around",
                  "overflow": "hidden",
                }
              }
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "FILMS LIST",
            "variant": "h2",
          },
          "ref": null,
          "rendered": "FILMS LIST",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(List)
              style={
                Object {
                  "display": "flex",
                  "flexWrap": "wrap",
                  "justifyContent": "space-around",
                  "overflow": "hidden",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [],
              "style": Object {
                "display": "flex",
                "flexWrap": "wrap",
                "justifyContent": "space-around",
                "overflow": "hidden",
              },
            },
            "ref": null,
            "rendered": Array [],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
